src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/EnhancedClinicalVariableCard.js

```diff
@@ -0,0 +1,140 @@
+import {
+  compose,
+  lifecycle,
+  setDisplayName,
+  withProps,
+  withPropsOnChange,
+  withState,
+} from 'recompose';
+import { connect } from 'react-redux';
+import {
+  isEmpty,
+  isEqual,
+} from 'lodash';
+
+import {
+  getSurvivalCurvesArray,
+  MAXIMUM_CURVES,
+} from '@ncigdc/utils/survivalplot';
+import { withTheme } from '@ncigdc/theme';
+
+import { makeDocCountInteger } from './helpers';
+import ClinicalVariableCard from './ClinicalVariableCard';
+
+export default compose(
+  setDisplayName('EnhancedSharedVariableCard'),
+  connect((state: any) => ({ analysis: state.analysis })),
+  withTheme,
+  withState('selectedBins', 'setSelectedBins', []),
+  withState('selectedSurvivalBins', 'setSelectedSurvivalBins', []),
+  withState('selectedSurvivalData', 'setSelectedSurvivalData', {}),
+  withState('selectedSurvivalLoadingIds', 'setSelectedSurvivalLoadingIds', []),
+  withState('survivalPlotLoading', 'setSurvivalPlotLoading', true),
+  withProps(({
+    fieldName,
+    filters,
+    setSelectedSurvivalData,
+    setSelectedSurvivalLoadingIds,
+    setSurvivalPlotLoading,
+    variable: { plotTypes },
+  }) => ({
+    updateSurvivalPlot: values => getSurvivalCurvesArray({
+      currentFilters: filters,
+      field: fieldName,
+      plotType: plotTypes,
+      values,
+    }).then(data => {
+      setSelectedSurvivalData(data);
+      setSurvivalPlotLoading(false);
+      setSelectedSurvivalLoadingIds([]);
+    }),
+  })),
+  withProps(
+    ({
+      selectedSurvivalBins,
+      setSelectedSurvivalBins,
+      setSelectedSurvivalLoadingIds,
+      setSurvivalPlotLoading,
+      survivalPlotValues,
+      survivalTableValues,
+      updateSurvivalPlot,
+      variable: { plotTypes }
+    }) => ({
+      populateSurvivalData: () => {
+        setSurvivalPlotLoading(true);
+        setSelectedSurvivalBins(survivalTableValues);
+        setSelectedSurvivalLoadingIds(survivalTableValues);
+        updateSurvivalPlot(survivalPlotValues);
+      },
+      updateSelectedSurvivalBins: (data, bin) => {
+        if (
+          selectedSurvivalBins.indexOf(bin.key) === -1 &&
+          selectedSurvivalBins.length >= MAXIMUM_CURVES
+        ) {
+          return;
+        }
+        setSurvivalPlotLoading(true);
+
+        const nextBins =
+          selectedSurvivalBins.indexOf(bin.key) === -1
+            ? selectedSurvivalBins.concat(bin.key)
+            : selectedSurvivalBins.filter(s => s !== bin.key);
+
+        setSelectedSurvivalBins(nextBins);
+        setSelectedSurvivalLoadingIds(nextBins);
+
+        const nextBinsForPlot = plotTypes === 'categorical' 
```

Comment:
Try to use prop from categorical or continuous card to represent the nextBins instead of checking the plotTypes.

Reply:
this would be a good improvement for when I have the selected survival bins stored in Redux, but right now it's not possible